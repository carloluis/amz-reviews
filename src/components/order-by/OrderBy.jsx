import React from 'react';
import PropTypes from 'prop-types';
import Select from '../select/Select';

const ORDERS = ['', 'Asc', 'Desc'];

const OrderBy = ({ value, orderBy }) => <Select onSelect={orderBy} title="Order by" selected={value} values={ORDERS} />;

OrderBy.propTypes = {
    orderBy: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired
};

export default OrderBy;
