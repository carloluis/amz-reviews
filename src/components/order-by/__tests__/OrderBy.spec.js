import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import OrderBy from '../OrderBy';

const PROPS = {
    orderBy: jest.fn(),
    value: ''
};

describe('<OrderBy/>', () => {
    it('snapshot', () => {
        const result = renderer.create(<OrderBy {...PROPS} />).toJSON();
        expect(result).toMatchSnapshot();
    });
});

describe('OrderBy', () => {
    it('should call the orderBy callback when selecting new value', () => {
        const wrapper = shallow(<OrderBy {...PROPS} />);
        const value = 'Asc';
        wrapper.simulate('select', value);
        expect(PROPS.orderBy).toHaveBeenCalledWith(value);
    });
});
