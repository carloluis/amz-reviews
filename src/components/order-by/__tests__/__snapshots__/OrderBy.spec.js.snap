// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderBy/> snapshot 1`] = `
<div
  className="container"
>
  <div
    className="input"
    onClick={[Function]}
  >
    Order by
     
    <i
      className="triangle"
    >
      â–¼
    </i>
  </div>
</div>
`;
